src/main/ruby/truffleruby/core/truffle/feature_loader.rb

```diff
@@ -198,11 +198,18 @@ def self.feature_provided?(feature, expanded)
     # MRI: loaded_feature_path
     # Search if $LOAD_PATH[i]/feature corresponds to loaded_feature.
     # Returns the $LOAD_PATH entry containing feature.
-    def self.loaded_feature_path(loaded_feature, feature, load_path)
+    def self.feature_path_loaded?(loaded_feature, feature, load_path)
       name_ext = extension(loaded_feature)
-      load_path.find do |p|
-        loaded_feature == "#{p}/#{feature}#{name_ext}" || loaded_feature == "#{p}/#{feature}"
-      end
+
+      path = if loaded_feature.end_with?("/#{feature}#{name_ext}")
+               loaded_feature.chomp("/#{feature}#{name_ext}")
```

Comment:
`chomp` will compare again (redundant), more efficient would be to use `byteslice` or `[]`